src/Lykke.AlgoStore.Services/AlgoStoreClientDataService.cs

```diff
@@ -256,6 +261,68 @@ public class AlgoStoreClientDataService : BaseAlgoStoreService, IAlgoStoreClient
             });
         }
 
+        /// <summary>
+        /// Create an algo from provided code
+        /// </summary>
+        /// <param name="clientId">Algo client Id</param>
+        /// <param name="clientName">Algo client name</param>
+        /// <param name="data">Algo data</param>
+        /// <param name="algoContent">Algo code content</param>
+        /// <returns></returns>
+        public async Task<AlgoClientMetaData> CreateAlgoAsync(string clientId, string clientName, AlgoMetaData data,
+            string algoContent)
+        {
+            return await LogTimedInfoAsync(nameof(CreateAlgoAsync), clientId, async () =>
+            {
+                if (string.IsNullOrWhiteSpace(clientId))
+                    throw new AlgoStoreException(AlgoStoreErrorCodes.ValidationError, "ClientId Is empty");
+
+                if(string.IsNullOrEmpty(algoContent))
+                    throw new AlgoStoreException(AlgoStoreErrorCodes.ValidationError, "Algo content is empty");
+
+                if (string.IsNullOrWhiteSpace(data.AlgoId))
+                    data.AlgoId = Guid.NewGuid().ToString();
+
+                if (!data.ValidateData(out var exception))
+                    throw exception;
+
+                //Validate algo code
+                var validationSession = _codeValidationService.StartSession(algoContent);
+                var validationResult = await validationSession.Validate();
+
+                if (!validationResult.IsSuccessful)
+                    throw new AlgoStoreException(AlgoStoreErrorCodes.ValidationError,
+                        $"Cannot save algo data. Algo code validation failed. ClientId: {clientId}, AlgoId: {data.AlgoId}");
+
+                //Extract algo metadata (parameters)
+                var extractedMetadata = await validationSession.ExtractMetadata();
+
+                data.AlgoMetaDataInformationJSON = JsonConvert.SerializeObject(extractedMetadata);
+
+                var clientData = new AlgoClientMetaData
+                {
+                    ClientId = clientId,
+                    Author = clientName,
```

Comment:
I think we can remove Author column or change it to ClinetId, regarding to https://lykkex.atlassian.net/wiki/spaces/LKEWALLET/pages/489160720/Working+with+personal+data.
Right now in Author column we save only the Id of the user